ampd/src/sui/json_rpc.rs

```diff
@@ -1,12 +1,14 @@
 use async_trait::async_trait;
 use ethers::providers::{JsonRpcClient, ProviderError};
+use mockall::automock;
 use sui_json_rpc_types::{SuiTransactionBlockResponse, SuiTransactionBlockResponseOptions};
 use sui_types::digests::TransactionDigest;
 
 use crate::json_rpc::Client;
 
 type Result<T> = error_stack::Result<T, ProviderError>;
 
+#[automock]
```

Comment:
Get the method to return `Result<Vec<Option<SuiTransactionBlockResponse>>>` as discussed.